javascript/javascript3/week1/script.js

```diff
@@ -0,0 +1,143 @@
+let weather = {
+    "apiKey": "fd548f55ee5a12f2b7b63bfc72cd0260",
+    fetchWeather: function (city) {
+        fetch(
+                "https://api.openweathermap.org/data/2.5/weather?q=" +
+                city +
+                "&units=metric&appid=" +
+                this.apiKey
+            )
+            .then((response) => response.json())
+            .then((data) => this.displayWeather(data));
+    },
+    fetchWeatherWithLocation: function () {
+        navigator.geolocation.getCurrentPosition(function (position) {
+            const lat = position.coords.latitude;
+            const lon = position.coords.longitude;
+            fetch(
+                    "https://api.openweathermap.org/data/2.5/weather?lat=" +
+                    lat + "&lon=" + lon +
+                    "&units=metric&appid=" +
+                    weather.apiKey
+                )
+                .then((response) => response.json())
+                .then((data) => weather.displayWeather(data));
+        });
+    },
+    searchWithJson: function () {
+        fetch('cities-of-turkey.json')
+            .then(response => response.json())
+            .then(data => {
+                const random = Math.floor(Math.random() * data.length);
+                const randomTrCity = data[random].name;
+                this.fetchWeather(randomTrCity);
+            });
+    },
+    displayWeather: function (data) {
+        const {
+            name
+        } = data;
+        const {
+            icon,
+            main,
+            description
+        } = data.weather[0];
```

Comment:
Because your fetch can potentially return nothing, this can result in an exception (index out of bounds)